qdrant-landing/content/articles/agentic-builders-guide.md

```diff
@@ -0,0 +1,168 @@
+Agentic memory systems like [mem0](https://docs.mem0.ai/components/vectordbs/dbs/qdrant) or [Cognee](https://docs.cognee.ai/examples/documentation-intelligence#scenario-intelligent-documentation-assistant-to-built-with-qdrant) are designed to give agents out-of-the-box long-term and short-term memory capabilities. You can easily drop in Qdrant as the vector search engine to power these systems. Qdrant powers the indexing, storing, and retrieval of data, letting the memory system focus on edits, summaries, and deciding what information to remember and what information can be forgotten.
+
+Agentic builders like [CrewAI](https://docs.crewai.com/en/tools/database-data/qdrantvectorsearchtool), [Dust](https://dust.tt/), [Voiceflow](https://www.voiceflow.com/), [Lyzr](https://www.lyzr.ai/), [n8n](https://n8n.io/), and [Cognigy](https://www.cognigy.com/) allow you to orchestrate your agentic flow. With these tools you can define multi-step, multi-agent workflows, and by adding Qdrant they gain a tool for grounding responses with your data, recalling past interactions, and searching multimodal knowledge. Qdrant provides the speed and precision needed for your workflow to be trustworthy as it scales.
+
+### **Evaluation Frameworks**
+
+Once you’ve built a fast, accurate, secure, and scalable agent, how do you know if it is actually working? How does it handle edge cases? What happens when things go wrong? You can’t improve what you don’t measure, which is where evaluations come in.
+
+#### [**Retrieval Quality**](https://qdrant.tech/documentation/beginner-tutorials/retrieval-quality/)
+
+Your agent’s ability to complete complex tasks is only as good as the context it can retrieve. It is crucial to closely and continuously monitor the agent’s performance with grounding checks to spot and prevent hallucinations, recall@k to ensure relevance in search, and MMR parameters for diversity.
+
+#### [**Performance-Cost Tradeoff**](https://qdrant.tech/documentation/guides/optimize/)
+
+In production agents, efficiency is one of, if not the, most important metric to track. First, in enterprise environments you must meet strict latency budgets. Evaluations also let you track the cost per task by tracking token usage and end-to-end compute time. Finally, you can track the effectiveness of your memory layer by monitoring cache hit rates for your memory banks.
+
+![Tradeoff Triangle](/articles_data/agentic-builders-guide/tradeoff-triangle.png)
+
+#### [**Guardrails & Fallbacks**](https://qdrant.tech/documentation/guides/security/)
+
+Just like humans, agents aren’t perfect. A production agentic system should expect and anticipate failures and have guardrails to handle them gracefully. You can also use a human in the loop when confidence scores are below a chosen threshold or the query touches on a high-stakes or sensitive topic. Qdrant also supports keyword search, which you can use as a fallback when semantic search doesn’t return the results your agent is looking for.
```

Comment:
I think it would be better to mention using hybrid search to cover all possible types of queries, not keyword search as a fallback.